### PR TITLE
Part two marcin

### PR DESCRIPTION
**Pull request by smolamarcin**

### CODE REVIEW

File: battleship-client/src/main/java/com/epam/solid/nie/client/Main.java [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 public class Main {
     public static void main(String[] args) {
-        ShipClient shipServer = new ShipSocketClient();
-        shipServer.run();
+        System.out.println("server ip is: " + (args.length > 0 ? args[0] : ""));
```
Comment (tqpz) [inline battleship-client/src/main/java/com/epam/solid/nie/client/Main.java]: replace it with logger

File: battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java [hunk 2]
```diff
@@ -8,8 +8,13 @@
...
 
+    private String hostName;
+
+    ShipSocketClient(String arg) {
+        hostName = arg;
+    }
+
     public void run() {
-        String hostName = "127.0.0.1";
         int portNumber = 8080;
```
Comment (tqpz) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java]: magic number

File: battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java [hunk 3]
```diff
@@ -21,12 +26,15 @@ public void run() {
             String fromServer;
             String fromUser;
             while ((fromServer = in.readLine()) != null) {
+                System.out.println("Server: " + fromServer);
```
Comment (tqpz) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java]: remove sout

File: battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java [hunk 4]
```diff
@@ -21,12 +26,15 @@ public void run() {
...
                 if (fromServer.equals("Bye."))
                     break;
 
                 fromUser = stdIn.readLine();
-                if (fromUser != null)
+                if (fromUser != null) {
                     out.println(fromUser);
+                    System.out.println("Server: " + fromUser);
```
Comment (tqpz) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java]: sout

File: battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java [hunk 5]
```diff
@@ -21,12 +26,15 @@ public void run() {
...
+                }
             }
         } catch (IOException ignored){
```
Comment (tqpz) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java]: exception unhandled

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/Controller.java [hunk 6]
```diff
@@ -0,0 +1,4 @@
+package com.epam.solid.nie.client.ui;
+
+public class Controller {
```
Comment (tqpz) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/Controller.java]: empty class should not been on master

File: battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java [hunk 7]
```diff
@@ -6,16 +6,25 @@
...
+    private final String ip;
+
+    ShipSocketServer(String ip) {
+        this.ip = ip;
+    }
 
     @Override
     public void run() {
-        try (ServerSocket serverSocket = new ServerSocket(portNumber, 0, InetAddress.getLoopbackAddress())) {
+        int portNumber = 8080;
```
Comment (tqpz) [inline battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java]: magic number

File: battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java [hunk 8]
```diff
@@ -6,16 +6,25 @@
...
+        try (
+                ServerSocket serverSocket = new ServerSocket(portNumber, 0, InetAddress.getByName(ip))
+        ) {
+            System.out.println("Server is here");
```
Comment (tqpz) [inline battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java]: sout

File: battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java [hunk 9]
```diff
@@ -6,16 +6,25 @@
...
             for (int nbOfPlayer = 0; nbOfPlayer < 2; nbOfPlayer++) {
                 Socket clientSocket = serverSocket.accept();
                 ServerThread serverThread = new ServerThread(clientSocket);
                 serverThread.start();
-                //System.out.println("registered");
+                System.out.println("registered");
```
Comment (tqpz) [inline battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java]: sout

File: battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java [hunk 10]
```diff
@@ -6,16 +6,25 @@
...
             }
         } catch (IOException ignored) {
```
Comment (tqpz) [inline battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java]: unhandled exception

File: battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java [hunk 11]
```diff
@@ -8,8 +8,13 @@
 
 class ShipSocketClient implements ShipClient {
 
+    private String hostName;
+
+    ShipSocketClient(String arg) {
```
Comment (tqpz) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java]: confusing argument name

File: battleship-client/src/main/resources/test.txt [hunk 12]
```diff
@@ -0,0 +1 @@
+sadasd
```
Comment (tqpz) [inline battleship-client/src/main/resources/test.txt]: unnecessary file

File: battleship-server/src/main/java/com.epam.solid.nie/server/ServerThread.java [hunk 13]
```diff
@@ -23,11 +23,11 @@ public void run() {
...
             Protocol protocol = new CommunicationProtocol();
-            outputLine = protocol.processInput("initial value");
+            outputLine = protocol.processInput("initial value", id);
             out.println(outputLine);
 
             while ((inputLine = in.readLine()) != null) {
-                outputLine = protocol.processInput(inputLine);
+                outputLine = protocol.processInput(inputLine, id);
                 out.println(outputLine);
                 if (outputLine.equals("Bye."))
```
Comment (tqpz) [inline battleship-server/src/main/java/com.epam.solid.nie/server/ServerThread.java]: hardcoded value

File: battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java [hunk 14]
```diff
@@ -6,16 +6,25 @@
...
 
     @Override
     public void run() {
-        try (ServerSocket serverSocket = new ServerSocket(portNumber, 0, InetAddress.getLoopbackAddress())) {
+        int portNumber = 8080;
+        try (
+                ServerSocket serverSocket = new ServerSocket(portNumber, 0, InetAddress.getByName(ip))
+        ) {
+            System.out.println("Server is here");
             for (int nbOfPlayer = 0; nbOfPlayer < 2; nbOfPlayer++) {
```
Comment (tqpz) [inline battleship-server/src/main/java/com.epam.solid.nie/server/ShipSocketServer.java]: messy code, imo this loop doesnt make sense

File: battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java (hunk 11 above)
Comment (Ozymadias) [reply battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java]: could you suggest better?

Comment (Ozymadias) [reply battleship-client/src/main/java/com/epam/solid/nie/client/ShipSocketClient.java]: arg can be changed to ip

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/Controller.java (hunk 6 above)
Comment (Ozymadias) [reply battleship-client/src/main/java/com/epam/solid/nie/client/ui/Controller.java]: to removed

File: battleship-server/src/main/java/com.epam.solid.nie/server/ServerThread.java (hunk 13 above)
Comment (Ozymadias) [reply battleship-server/src/main/java/com.epam.solid.nie/server/ServerThread.java]: gameOverMessage = "Bye."

**Pull request closed without merging**